### DISCUSSION TITLE
Prompt for password in `ente-decrypt` binary

### DISCUSSION THREAD

**Discussion by ziadoz** [ente-io/ente#669] [Enhancements]
It would be nice if the `ente-decrypt` binary prompted the user for their decryption password, instead of it being a positional parameter, which will be recorded in your Bash/ZSH history.

**Before:**
```
ente-decrypt in.txt <password> out.txt
```

**After:**
```
ente-decrypt in.txt out.txt

Please enter your password:
```

In the short term I've added the entire command to my history ignores to avoid password leakage:

```
export HISTORY_IGNORE='(ente-decrypt*)'
```

**Comment by mnvr** [comment]
Not to take away from this feature request (I also feel it's a valid use case) but a common thing that I do is start a command with a space. On the two shells that I've used recently (bash and zsh), this prevents the command from being saved into history.

**Reply by ziadoz** [reply]
@mnvr Ah, I didn't know that. I'll remember that in future. 👍

**Comment by ua741** [comment]
Thank you for flagging this issue. We have deprecated this tool and added this functionality inside the **Ente CLI** tool.

You can download the latest build from [here](https://github.com/ente-io/ente/releases?q=tag%3Acli-v0&expanded=true), and use the following command to decrypt.
```bash
./ente auth decrypt in.txt out.txt
```
